src/js/module.js

```diff
@@ -149,10 +153,13 @@ iotjs_module_t._resolveFilepath = function(id, root) {
     var pkgMainFile = JSON.parse(pkgSrc).main;
 
     // pkgmain[.ext]
-    if (filepath = tryPath(modulePath + '/' + pkgMainFile, ext)) {
+    if (filepath = tryPath(modulePath + '/' + pkgMainFile, exts[ext_index])) {
       return filepath;
     }
   }
+  if (ext_index++ < exts.length) {
```

Comment:
prevent in-line `++` statement whenever possible.

Reply:
Is this to consider performance or what?

Reply:
Readability IMO.